downloads.html

```diff
@@ -109,16 +109,16 @@ <h3 class="download-version">3.5.2<a href="#3.5.2"><i class="fas fa-link " style
             Released Dec 11, 2023
         </li>
         <li>
-            <a href="https://downloads.apache.org/kafka/3.5.2/RELEASE_NOTES.html">Release Notes</a>
+            <a href="https://archive.apache.org/dist/kafka/3.5.2/RELEASE_NOTES.html">Release Notes</a>
```

Comment:
3.5 branch is not EOL yet. Why are we moving to archive for this?

Reply:
I don't have an objection but I want to ensure that we have some fixed criteria on what to move to archive and what not to. I agree it's unlikely that we will have a bug fix version of 3.5.x until it is EOL in June but technically, it's still not EOL'ed.

I am ok with the criteria being, best judgement of the PMC member at the time of release of a minor version. My motivation is to ensure that we have a criteria which all PMC members could follow in future.

> We have a tendency to forget to clean it up

Is there something we can do in the release process to enforce this? I was thinking of having a multi-stage script which is run by PMC member and it will prompt whether this is done or not (or automatically do it). We can take the help of community to improve this as part of https://issues.apache.org/jira/browse/KAFKA-15198

Reply:
My understanding is that we use "best judgment" to select what to delete so it may vary each time. Maybe deciding to keep all not EOL'ed releases in the mirroring system would be a better process. 
I'll undo the 3.5.2 changes and will keep this release in the mirroring system for now.

> Is there something we can do in the release process to enforce this?
Some steps have to be performed by PMC members and once a release is announced it's easy to move onto something else and not complete the post release steps. If possible, yes we should try to automate as much as possible but it's often tricky.

Reply:
I've undone the changes related to 3.5.2.